Addition of Metabolomics Workbench related APIs.

mano-at-sdsc opened the pull request:
These include APIs for the gene ID conversion and MetGENE.

mano-at-sdsc committed:
Addition of Metabolomics Workbench related APIs.
These include APIs for the gene ID conversion and MetGENE.

mano-at-sdsc committed:
Merge remote-tracking branch 'upstream/main' into playbook-partnership-mano-20221129

u8sand committed:
Move MW work to subdirectory

u8sand committed:
Migrate io-ts to zod

u8sand committed:
Migrate GeneSymbol to GeneTerm

u8sand committed:
Remove duplicate MetGeneStudies

u8sand committed:
Fixed an issue with set to array conversion

mano-at-sdsc committed:
Merge pull request #1 from nih-cfde/mw

My Fixes to your PR

u8sand commented on components/MW/ConvertedGeneID/index.tsx:
```diff
@@ -0,0 +1,109 @@
+  datastr = datastr.replace("^[","");
+  datastr = datastr.replace("]$","");
+
+  const dataobj = JSON.parse(datastr);
```
I'm not entirely certain what's going on here but if you're removing `^[` and `]$` from a json stringified data and re-parsing it then perhaps you're simply removing the outer list but this would only be valid if the you only have **one** list.. in which case shouldn't the code just be: `const dataobj = data[0]` ? or more elegantly `const [dataobj] = data`

mano-at-sdsc replied:
OK. I have now: const [dataobj] = data; I have left the original lines as comment on a s single line

u8sand commented on components/MW/ConvertedGeneID/index.tsx:
```diff
@@ -0,0 +1,109 @@
+  })
+  // this should have a codec which can encode or decode the data type represented by this node
+  //  using zod, a compile-time and runtime type-safe codec can be constructed
+  .codec(z.any())
```
Please define the schema of your object, z.any() is fine for prototyping but it's not suitable for production.

mano-at-sdsc replied:
I am not understanding this part. Where and how do I define the schema of the object (data?)?

u8sand replied:
Here are some examples of other schemas:
https://github.com/nih-cfde/playbook-partnership/blob/main/components/service/mygeneinfo/index.tsx#L20-L29

https://github.com/nih-cfde/playbook-partnership/blob/main/components/core/significant_tissues/index.tsx#L5-L8

If you have an example of the data you can also use this:
https://rsinohara.github.io/json-to-zod-react/

To get you started -- it's basically the "shape" of your data.

mano-at-sdsc replied:
We will fix this in the next round; For now, I have made my edits directly into the files here on github, e.g. in
https://github.com/metabolomicsworkbench/playbook-partnership/edit/playbook-partnership-mano-20221129/components/MW/geneidconv/index.tsx

Sumana is doing the same.

mano-at-sdsc replied:
We have resolved all but three (all related to z.any), whee we need help. Can we do that in the next round after merging with nih-cfde:main, so that we can clone again and actually test locally while making the changes.

u8sand replied:
@mano-at-sdsc this branch is up to date with main so you *can* test it locally.. you need to `git fetch && git pull` it into your local workspace.

mano-at-sdsc replied:
Thanks, @u8sand [keeping @susrinivasan in the loop]. I see so many branches on my local system. Can I start with a git clone. I got some warning/error in trying git fetch && git pull.

[mano@sc-cfdewebdev playbook-partnership]$ git fetch && git pull
remote: Enumerating objects: 90, done.
remote: Counting objects: 100% (48/48), done.
remote: Compressing objects: 100% (17/17), done.
remote: Total 90 (delta 38), reused 31 (delta 31), pack-reused 42
Unpacking objects: 100% (90/90), 18.51 KiB | 210.00 KiB/s, done.
From https://github.com/metabolomicsworkbench/playbook-partnership
   0e19dec..c6480bd  playbook-partnership-mano-20221129 -> origin/playbook-partnership-mano-20221129
 * [new branch]      main                               -> origin/main
Updating 0e19dec..c6480bd
error: Your local changes to the following files would be overwritten by merge:
        components/ConvertedGeneID/index.tsx
Please commit your changes or stash them before you merge.
Aborting
[mano@sc-cfdewebdev playbook-partnership]$

u8sand replied:
>Please commit your changes or stash them before you merge.

```
git add --all .
git stash
```

Then you can pull. This is because your workspace has un-committed changes that would be overwritten if you were to pull.. (as the error states) If you care about your workspace changes afterwards you can run `git stash pop`

mano-at-sdsc replied:
Thanks, @u8sand. It worked and I see the UI. Giving some issue re reading json object, but I will debug that or ask you if I have trouble. @susrinivasan  will try the same. In her case, the local may not be pointing to playbook-partnership-mano-20221129 [ i.e., https://github.com/metabolomicsworkbench/playbook-partnership/tree/playbook-partnership-mano-20221129], so, does she need to specify from where to fetch/pull, as in

git fetch https://github.com/metabolomicsworkbench/playbook-partnership/tree/playbook-partnership-mano-20221129

u8sand replied:
@mano-at-sdsc As long as she has the metabolomicsworkbench/playbook-partnership configured as a remote, she can check it out in a branch
```
git fetch remotename
git checkout -b somebranch -t remotename/somebranch
```

mano-at-sdsc replied:
@u8sand I added the schema and did
git add --all .
git stash

then

git fetch upstream
git pull

Now I don't see my latest changes in my file from today. How do I revert to the state before I did "git fetch upstream" and "git pull". If not possible, OK, I can just retype.

u8sand replied:
No, @mano-at-sdsc, your changes are in the `stash` since you ran `git stash` you can get them back with `git stash pop`

mano-at-sdsc replied:
Thanks @u8sand . I got back my edits from today. Now, I want to bring in any recent changes from nih-cfde/playbook-partnership to my local copy in the current branch.
C[mano@sc-cfdewebdev playbook-partnership]$ git remote -v
mano-at-sdsc/playbook-partnership       https://github.com/mano-at-sdsc/playbook-partnership (fetch)
mano-at-sdsc/playbook-partnership       https://github.com/mano-at-sdsc/playbook-partnership (push)
origin  https://github.com/metabolomicsworkbench/playbook-partnership.git (fetch)
origin  https://github.com/metabolomicsworkbench/playbook-partnership.git (push)
upstream        https://github.com/nih-cfde/playbook-partnership/ (fetch)
upstream        https://github.com/nih-cfde/playbook-partnership/ (push)

I suppose origin actually refers to
https://github.com/metabolomicsworkbench/playbook-partnership/tree/playbook-partnership-mano-20221129
even though it just says https://github.com/metabolomicsworkbench/playbook-partnership.git since the local active branch is playbook-partnership-mano-20221129.

I have taken a manual backup of my files. I suppose I should do:

git add --all .
git stash

git fetch upstream
git merge upstream/main

Please let me know. Thanks.

u8sand replied:
@mano-at-sdsc you seem to be conflating two different things, the remote (a repository) and the branch (just a *pointer* in the repository). `fetch` will download the repository, `pull` will update your current branch to incorporate things from the remote tracking branch (usually the same name in different repos. i.e. main vs origin/main)

Unfortunately it's unclear to me which changes you are trying to incorporate, there are no real changes necessary to take from the nih-cfde/playbook-partnership repo (which is where upstream points). This branch exists on github at metabolomicsworkbench:playbook-partnership-mano-20221129 which is equivalently `origin/playbook-partnership-mano-20221129` from the perspective of your local repo.

You should make sure you're on that branch `git checkout playbook-partnership-mano-20221129 ` and you should be able to simply `git pull origin`.

It will likely be easier to schedule a call than me try to understand what you're trying to do in these comments..

mano-at-sdsc replied:
I agree. I will sync with Sumana as well and then schedule something to make sure we are on the right track. Just now, following what we did last time, I pushed local changes to origin/playbook-partnership-mano-20221129 and I see it on github: https://github.com/metabolomicsworkbench/playbook-partnership/tree/playbook-partnership-mano-20221129. Then, when I did git fetch upstream && git merge upstream, it seems to have incorporated some changes related to glygen from nih-cfde/playbook-partnership into my local copy [looks like I didn't have gly_gen folder in my local copy earlier]. So far, it seems to have gone as expected. Thanks.

mano-at-sdsc replied:
Hi @u8sand . We have pushed the updated metgene files as well (schema added and codec argument set correctly) to  https://github.com/metabolomicsworkbench/playbook-partnership/tree/playbook-partnership-mano-20221129. Would you like to meet over zoom sometime, may be next week, where we can walk over the recent changes, or you can review it yourself. I think these are now ready to be integrated into your master repo. Thanks.

u8sand commented on components/MW/ConvertedGeneID/index.tsx:
```diff
@@ -0,0 +1,109 @@
+    datastr = datastr.replace("^[","");
+    datastr = datastr.replace("]$","");
+
+    const dataobj = JSON.parse(datastr);
```
Same comment from before about `const [dataobj] = data`

mano-at-sdsc replied:
Done

mano-at-sdsc replied:
What seems to work is:
const dataobj = [data][0];

u8sand replied:
[data][0] doesn't do anything at all. In this case dataobj is equal to data..

mano-at-sdsc replied:
Yes. Thanks. Now, I am using data directly; may be things were not working for some other reason.

u8sand commented on components/MW/ConvertedGeneID/index.tsx:
```diff
@@ -0,0 +1,109 @@
+    for (var i = 0; i < uniqENTREZID.length; i++) {
+      stru = stru + `<a href='https://www.ncbi.nlm.nih.gov/gene/${uniqENTREZID[i]}'>${uniqSYMBOL[i]}</a> ${uniqGENENAME[i]}`;
+      //str1[i] = `<a href="https://www.ncbi.nlm.nih.gov/gene/${ENTREZID[i]}">${SYMBOL[i]}</a> ${GENENAME[i]}`;
+    }
```
You create these variables but they aren't used anywhere?

mano-at-sdsc replied:
Deleted such lines now

u8sand commented on components/MW/ConvertedGeneID/index.tsx:
```diff
@@ -0,0 +1,109 @@
+    //})}
+      //junk
+      //<div>        <a href={`https://www.ncbi.nlm.nih.gov/gene/${geneinfo.entrezgene}`}>{geneinfo.symbol}</a> {geneinfo.name}      </div>
+    ) //; };
```
Please remove these extra unused comments

mano-at-sdsc replied:
Done

u8sand commented on components/MW/ConvertedGeneID/index.tsx:
```diff
@@ -0,0 +1,109 @@
+      //<div><h6>{str1.join(" ; ")}</h6></div>
+      //{stru.map((val, key) => { 
+      //    return (
+      //      <div><h6><table><tr><td>{stru}</td></tr></table></h6></div> // ask Daniel why this str is not rendering as a web page/url
```
The "answer" to this comment is that you should use JSX rather than a string of raw HTML.

For example, here `stru` would probably be something along the lines of:
```jsx
<div>
  {uniqENTREZID.map((_, i) => (
    <div key={id}>
      <a href={`https://www.ncbi.nlm.nih.gov/gene/${uniqENTREZID[i]}`}>{uniqSYMBOL[i]}</a> {uniqGENENAME[i]}
    </div>
  ))}
</div>
```

Note that you build the HTML-like JSX syntax *inline*.

mano-at-sdsc replied:
Thanks.

u8sand commented on components/MW/ConvertedGeneID/package.json:
```diff
@@ -0,0 +1,14 @@
+  "name": "ConvertedGeneID",
+  "version": "1.0.0",
+  "license": "CC-BY-NC-SA-4.0",
+  "author": "",
```
Please add the author

u8sand commented on components/MW/geneidconv/index.tsx:
```diff
@@ -0,0 +1,46 @@
+    const res = await req.json()
+
+    //return props.inputs.gene
+    return await res
```
The additional "await" should not be here, `res` was already awaited.

u8sand commented on components/MW/geneidconv/package.json:
```diff
@@ -0,0 +1,14 @@
+  "name": "geneidconv",
+  "version": "1.0.0",
+  "license": "CC-BY-NC-SA-4.0",
+  "author": "",
```
Add Author

u8sand commented on components/MW/metaboliteInfo/package.json:
```diff
@@ -0,0 +1,14 @@
+  "name": "myidentity",
+  "version": "1.0.0",
+  "license": "CC-BY-NC-SA-4.0",
+  "author": "",
```
Add Author

susrinivasan replied:
Done

u8sand commented on components/MW/metaboliteInfo/package.json:
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "myidentity",
```
Please fix the name of this component

susrinivasan replied:
The name of the component is MetaboliteInfo. Will fix that. Can I do it here directly?

u8sand replied:
GitHub doesn't really have such a feature but once you've addressed the change on your code, committed and pushed to the branch which this PR was initiated from (`metabolomicsworkbench:playbook-partnership-mano-20221129`), this comment will be "resolved" automatically.

susrinivasan replied:
Done

u8sand commented on components/MW/metaboliteInfo/index.tsx:
```diff
@@ -0,0 +1,26 @@
+    const req = await fetch(`https://www.metabolomicsworkbench.org/rest/refmet/name/${metName}/all`)
+    const res = await req.json()
+    
+    return await res
```
Redundant await here

susrinivasan replied:
You mean line #23 is just "return res"?

u8sand replied:
Yes that's right

susrinivasan replied:
Done

u8sand commented on components/MW/metabolite_summary/index.tsx:
```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import { MetaNode } from '@/spec/metanode'
+import { z } from 'zod'
+//import { valHooks } from 'jquery'
```
Remove the deadcode in comments

susrinivasan replied:
Do I create a pull request for each change I make by editing the file on github?

u8sand replied:
No, when you make a commit, and push it to the branch, it will automatically add the changes to this existing pull request.

susrinivasan replied:
Done

u8sand commented on components/MW/metabolite_summary/package.json:
```diff
@@ -0,0 +1,14 @@
+  "name": "metabolite_summary",
+  "version": "1.0.0",
+  "license": "CC-BY-NC-SA-4.0",
+  "author": "",
```
Add Author

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_metabolite_table/index.tsx:
```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import { MetaNode } from '@/spec/metanode'
+import { z } from 'zod'
+import { valHooks } from 'jquery'
```
You don't use this anywhere, please remove it

susrinivasan replied:
You want me to remove lines 3 and 4? or only line 3?  I have no idea how they appeared there. Please confirm.

u8sand replied:
Just line 4. `zod` is used later for the schema definition (it was previously `io-ts`)

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_metabolite_table/index.tsx:
```diff
@@ -0,0 +1,64 @@
+          {data.map((val, key) => {
+          var all_rxn_ids = val.KEGG_REACTION_ID
+          var cpd_id = val.KEGG_COMPOUND_ID     
+          var rxn_id_arr = all_rxn_ids.split(", ")
```
Please replace `var`s with `const` or  with `let` if you need reassignment. `var` can have odd scoping behaviors

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_metabolite_table/index.tsx:
```diff
@@ -0,0 +1,64 @@
+            <tr key={key}>
+              <td><a href = {`https://www.kegg.jp/entry/${val.KEGG_COMPOUND_ID}`} target = "__blank">{val.KEGG_COMPOUND_ID}</a></td>
+              <td><a href = {`https://www.metabolomicsworkbench.org/databases/refmet/refmet_details.php?REFMET_NAME=${val.REFMET_NAME}`} target = "_blank">{val.REFMET_NAME}</a></td>
+              <td>{rxn_urls}</td>
```
Your list elements are missing `key=`, instead I recommend doing this inline eliminating most of the steps before the return statement:

```tsx
<td>
  {rxn_id_arr.map((rxn_id, i) =>
    <a key={i} href = {`https://www.genome.jp/entry/rn:${rxn_id}`} target = "_blank">{rxn_id}</a>
  )}
</td>
```

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_metabolite_table/index.tsx:
```diff
@@ -0,0 +1,64 @@
+  })
+  // this should have a codec which can encode or decode the data type represented by this node
+  //  using zod, a compile-time and runtime type-safe codec can be constructed
+  .codec(z.any())
```
Please add the runtime type of this, z.any() is not suitable for production

susrinivasan replied:
Would the runtime type be MetgeneMetaboliteTable in this case?

u8sand replied:
I can only guess it by the code, you have the best ability to figure this out since you are more familiar with the API:

Given the code you have here:
```tsx
data.map((val, key) => {
          const all_rxn_ids = val.KEGG_REACTION_ID
          const cpd_id = val.KEGG_COMPOUND_ID     
          const rxn_id_arr = all_rxn_ids.split(", ")
 // ...
})
```

I assume it looks something like
```tsx
z.array(z.object({
  KEGG_REACTION_ID: z.string(),
  KEGG_COMPOUND_ID: z.string(),
}))
```

u8sand commented on components/MW/metgene_metabolite_table/package.json:
```diff
@@ -0,0 +1,14 @@
+  "name": "metgene_metabolite_table",
+  "version": "1.0.0",
+  "license": "CC-BY-NC-SA-4.0",
+  "author": "",
```
Add author

susrinivasan replied:
Done

u8sand replied:
When done you can just click the "Resolve conversation" button :)

u8sand commented on components/MW/metgene_metabolites/index.tsx:
```diff
@@ -0,0 +1,33 @@
+    
+
+    //return props.inputs.input
+    return await res
```
Remove redundant await

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_metabolites/index.tsx:
```diff
@@ -0,0 +1,33 @@
+  .resolve(async (props) => {
+    const species_id = "hsa"
+    const geneID_type = "SYMBOL"
+    //const gene_ID = "3098"
```
Remove deadcode comments

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_metabolites/package.json:
```diff
@@ -0,0 +1,14 @@
+  "name": "metgene_metabolites",
+  "version": "1.0.0",
+  "license": "CC-BY-NC-SA-4.0",
+  "author": "",
```
Add Author

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_rxn_table/index.tsx:
```diff
@@ -0,0 +1,47 @@
+  })
+  // this should have a codec which can encode or decode the data type represented by this node
+  //  using zod, a compile-time and runtime type-safe codec can be constructed
+  .codec(z.any())
```
Please add schema for your type

susrinivasan replied:
How do I add schema for type MetGeneRxnTable ? I need help in fixing this.

u8sand commented on components/MW/metgene_rxn_table/index.tsx:
```diff
@@ -0,0 +1,47 @@
+    const heading3 = "Kegg Rxn Name"
+    
+    return (
+      //<div>{JSON.stringify(data)}</div>
```
Remove dead-code comments

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_rxn_table/package.json:
```diff
@@ -0,0 +1,14 @@
+  "name": "metgene_rxn_table",
+  "version": "1.0.0",
+  "license": "CC-BY-NC-SA-4.0",
+  "author": "",
```
Add author

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_rxns/index.tsx:
```diff
@@ -0,0 +1,33 @@
+  .resolve(async (props) => {
+    const species_id = "hsa"
+    const geneID_type = "SYMBOL"
+    //const gene_ID = "3098"
```
Remove dead-code comments

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_rxns/index.tsx:
```diff
@@ -0,0 +1,33 @@
+    
+
+    //return props.inputs.input
+    return await res
```
Remove redundant await

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_rxns/package.json:
```diff
@@ -0,0 +1,14 @@
+  "name": "metgene_rxns",
+  "version": "1.0.0",
+  "license": "CC-BY-NC-SA-4.0",
+  "author": "",
```
Add author

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_studies/index.tsx:
```diff
@@ -0,0 +1,35 @@
+    
+
+    //return props.inputs.input
+    return await res
```
Remove redundant await

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_studies/index.tsx:
```diff
@@ -0,0 +1,35 @@
+  .resolve(async (props) => {
+    const species_id = "hsa"
+    const geneID_type = "SYMBOL"
+    //const gene_ID = "3098"
```
Remove dead-code comments

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_study_table/index.tsx:
```diff
@@ -0,0 +1,61 @@
+    const heading2 = "Refmet Name"
+    const heading3 = "Study Ids"
+    const refmet_url = "\""
+    const url_close = "\""
```
Remove unused variables

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_study_table/index.tsx:
```diff
@@ -0,0 +1,61 @@
+            <tr key={key}>
+              <td>{val.KEGG_COMPOUND_ID}</td>
+              <td><a href = {`https://www.metabolomicsworkbench.org/databases/refmet/refmet_details.php?REFMET_NAME=${val.REFMET_NAME}`} target = "_blank">{val.REFMET_NAME}</a></td>
+              <td>{urls}</td>
```
See above, use inline JSX `.map`

susrinivasan replied:
Done

u8sand commented on components/MW/metgene_study_table/package.json:
```diff
@@ -0,0 +1,14 @@
+  "name": "metgene_study_table",
+  "version": "1.0.0",
+  "license": "CC-BY-NC-SA-4.0",
+  "author": "",
```
Add author

susrinivasan replied:
Done

u8sand commented on components/MW/ConvertedGeneID/index.tsx:
```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+import { MetaNode } from '@/spec/metanode'
+import { z } from 'zod'
+import { VALID_LOADERS } from 'next/dist/shared/lib/image-config'
```
Remove this unused import

u8sand reviewed (commented):
Thank you for these changes, it's looking pretty good. However, there are a number of things which should be addressed before we merge it into the main branch and I've left comments throughout your PR about what should be fixed.

The common themes are:
- missing author in many of the component package.json's
- missing codecs on your Data type MetaNodes which are essential for catching errors that could occur in production
- some code cleanup related to typescript/react best practices & dead-code comment reduction

Finally, a question was asked about using a set of genes instead of a gene term, this can be done by using

```diff
-import { GeneTerm } from '@/components/core/input/term'
+import { GeneSet } from '@/components/core/input/set'
```

And of course updating it in the `inputs` and finally dealing with the fact, in your code, that it's a set `["Gene1", "Gene2", ...]` instead of a term like `"Gene1"`.

Please let me know if I can help or clarify anything.

susrinivasan committed:
Update package.json

Changed to MetaboliteInfo

susrinivasan committed:
Update index.tsx

Changed "return await res" to "return res"

susrinivasan committed:
Update index.tsx

Deleted dead code in comments in the import.

susrinivasan committed:
Update package.json

Updated author information.

susrinivasan committed:
Update index.tsx

Changed var to const and let. Haven't changed var within for loop though. Unsure whether that needs to be changed as well.

susrinivasan committed:
Update index.tsx

Deleted the line
 import { valHooks } from 'jquery'

susrinivasan committed:
Update package.json

Updated author information.

susrinivasan committed:
Update index.tsx

Removed redundant await and changed to return res in line 31.

susrinivasan committed:
Update index.tsx

Deleted commented line of dead code.

mano-at-sdsc committed:
Update index.tsx

mano-at-sdsc committed:
Update index.tsx

susrinivasan committed:
Update package.json

Updated author information.

mano-at-sdsc committed:
Update package.json

mano-at-sdsc committed:
Update package.json

susrinivasan committed:
Update index.tsx

Removed dead code comments.

susrinivasan committed:
Update package.json

Updated author information

mano-at-sdsc committed:
Update package.json

susrinivasan committed:
Update package.json

Updated author information.

susrinivasan committed:
Update index.tsx

Removed dead code and redundant await.

susrinivasan committed:
Update package.json

Updated author info.

susrinivasan committed:
Update package.json

Updated author info

susrinivasan committed:
Update index.tsx

Removed dead code and redundant await.

susrinivasan committed:
Update index.tsx

Removed unused 
const refmet_url = "\""
    const url_close = "\""

susrinivasan committed:
Update package.json

Added author info.

susrinivasan committed:
Update package.json

Updated author.

susrinivasan committed:
Update index.tsx

Changed reaction IDs wrapping to inline.

susrinivasan committed:
Update index.tsx

Removed dead code comments.

susrinivasan committed:
Update index.tsx

Changed study ID URLS to inline.

mano-at-sdsc committed:
Update index.tsx

mano-at-sdsc committed:
Update index.tsx

mano-at-sdsc committed:
Merge pull request #24 from susrinivasan/patch-23

Update index.tsx

mano-at-sdsc committed:
Merge pull request #23 from susrinivasan/patch-22

Update index.tsx

mano-at-sdsc committed:
Merge pull request #22 from susrinivasan/patch-21

Update index.tsx

mano-at-sdsc committed:
Merge pull request #21 from susrinivasan/patch-20

Update package.json

mano-at-sdsc committed:
Merge pull request #20 from susrinivasan/patch-19

Update package.json

susrinivasan committed:
Merge pull request #2 from susrinivasan/patch-1

Update package.json

susrinivasan committed:
Merge pull request #3 from susrinivasan/patch-2

Update index.tsx

mano-at-sdsc committed:
Merge pull request #19 from susrinivasan/patch-18

Update index.tsx

susrinivasan committed:
Merge pull request #4 from susrinivasan/patch-3

Update index.tsx

susrinivasan committed:
Merge pull request #5 from susrinivasan/patch-4

Update package.json

susrinivasan committed:
Merge pull request #18 from susrinivasan/patch-17

Update index.tsx

susrinivasan committed:
Merge pull request #17 from susrinivasan/patch-16

Update package.json

susrinivasan committed:
Merge pull request #16 from susrinivasan/patch-15

Update package.json

susrinivasan committed:
Merge pull request #15 from susrinivasan/patch-14

Update index.tsx

susrinivasan committed:
Merge pull request #14 from susrinivasan/patch-13

Update package.json

susrinivasan committed:
Merge pull request #13 from susrinivasan/patch-12

Update package.json

susrinivasan committed:
Merge pull request #11 from susrinivasan/patch-10

Update package.json

susrinivasan committed:
Merge pull request #7 from susrinivasan/patch-6

Update index.tsx

susrinivasan committed:
Merge pull request #8 from susrinivasan/patch-7

Update package.json

susrinivasan committed:
Merge pull request #9 from susrinivasan/patch-8

Update index.tsx

susrinivasan committed:
Merge pull request #10 from susrinivasan/patch-9

Update index.tsx

susrinivasan committed:
Merge branch 'playbook-partnership-mano-20221129' into patch-5

susrinivasan committed:
Merge pull request #6 from susrinivasan/patch-5

Update index.tsx

susrinivasan committed:
Merge branch 'playbook-partnership-mano-20221129' into patch-11

susrinivasan committed:
Merge pull request #12 from susrinivasan/patch-11

Update index.tsx

mano-at-sdsc committed:
Updated ConvertedGeneID index.tsx
Now, using data variable/object directly.

mano-at-sdsc committed:
Merge branch 'playbook-partnership-mano-20221129' of https://github.com/metabolomicsworkbench/playbook-partnership into playbook-partnership-mano-20221129

mano-at-sdsc committed:
ConvertedGeneID index.tsx simplified
ConvertedGeneID index.tsx simplified: removed redundant code

susrinivasan committed:
Added table rendering nested maps for metabolites

susrinivasan committed:
Merge branch 'playbook-partnership-mano-20221129' of https://github.com/metabolomicsworkbench/playbook-partnership into playbook-partnership-mano-20221129

susrinivasan committed:
Added support for multiple genes

mano-at-sdsc committed:
Added the schema to be used with ConvertedGeneID, for the json object returned by the REST API

mano-at-sdsc committed:
Merge remote-tracking branch 'upstream' into playbook-partnership-mano-20221129

mano-at-sdsc committed:
Added schema for the json ouputs from metgene APIs for reaction, metabolite and studies

mano-at-sdsc committed:
Add schema for MW REST json output for metabolite_summary

u8sand commented on components/MW/metgene_study_table/index.tsx:
```diff
@@ -0,0 +1,74 @@
+                    sep_char = "";
+                  }
+                  return (
+                  <a key={i} href = {`https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&StudyID=${study_id}`} target = "_blank">{study_id}{sep_char}</a>
```
How about
```tsx
{study_id_arr.map((study_id: string, i: number) =>
  <>
    {i > 0 ? <span>, </span> : null}
    <a key={i} href = {`https://www.metabolomicsworkbench.org/data/DRCCMetadata.php?Mode=Study&StudyID=${study_id}`} target = "_blank">{study_id}</a>
  </>
)}
```

u8sand commented on components/MW/metaboliteInfo/package.json:
```diff
@@ -0,0 +1,14 @@
+  "name": "MetaboliteInfo",
+  "version": "1.0.0",
+  "license": "CC-BY-NC-SA-4.0",
+  "author": "",
```
Missing author still?

mano-at-sdsc replied:
Thanks, fixing it. For "main", "main": "index.tsx" or "main": "index.ts". Does it matter.

u8sand replied:
It depends on which file you actually have next to it in the same directory if it's `index.ts` or `index.tsx`. I think they probably should all be `index.tsx` -- i don't currently rely on this for anything but it would be  good to have it accurate.

mano-at-sdsc replied:
It's all index.tsx. I also did the <span> logic in metabolite and studies table. Should be good to go now. Thanks a lot.

u8sand reviewed (changes requested):
All main concerns were addressed besides a missing author still. But it seems to build successfully now so it's basically ready.

I left a suggestion for a slight visual improvement to the part where you try and intersperse commas which you can incorporate if you like, once the author is added I can merge this.

mano-at-sdsc committed:
Added author, email, in metgene, and better streamlined yse of , in study or reaction list
